winpr/libwinpr/clipboard/posix.c

```diff
@@ -627,20 +638,18 @@ static void* convert_filedescriptors_to_uri_list(wClipboard* clipboard, UINT32 f
 
 	for (x = 0; x < count; x++)
 	{
+		if (_wcschr(descriptors[x].cFileName, L'\\') != NULL)
+		{
+			continue;
+		}
 		int rc;
 		const FILEDESCRIPTORW* cur = &descriptors[x];
 		size_t curLen = _wcsnlen(cur->cFileName, ARRAYSIZE(cur->cFileName));
 		char* curName = NULL;
 		rc = ConvertFromUnicode(CP_UTF8, 0, cur->cFileName, (int)curLen, &curName, 0, NULL, NULL);
 
-		if (rc != (int)curLen)
-		{
-			free(curName);
-			free(dst);
-			return NULL;
-		}
-
-		rc = _snprintf(&dst[pos], alloc - pos, "%s/%s\r\n", clipboard->delegate.basePath, curName);
+		rc = _snprintf(&dst[pos], alloc - pos, "file://%s/%s\n", clipboard->delegate.basePath,
+		               curName);
```

Comment:
I don't know why won't be able to . I've tested with  filenames with ä, ö, ü chars. You can copy them properly.

Reply:
I just looked up my test case for this (I implemented a server side cliprdr solution where I ran into this issue too):

On a MS Windows machine create directory called `#ä` and into that directory another directory, called `äö#`. When copying the `#ä` directory and pasting it locally, you won't get the expected result (don't remember what went wrong there, only that it didn't work).  
If the uris on the uri-list are escaped, everything worked fine.

Reply:
> directory called `#ä`

Oh... I believe the issue here is not “ä” but “#”, which is special in URLs but allowed in Windows filenames just fine. Trying to use “&” in for example should have similar issues if that's the case.

I guess “ä” are fine without escaping because the parser does not particularly care about them and skips them as is, but for “#” it does try to interpret the rest of the string as anchor, and that breaks filenames.

Reply:
Oh i haven't seem this . still more works to be done. like `#` `?`  :disappointed: 

ref:https://en.wikipedia.org/wiki/File_URI_scheme

Reply:
Maybe fixed.  Haven't had enough testing.